enterprise/tools/check_metrics/check_metrics.go

```diff
@@ -0,0 +1,134 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"flag"
+	"fmt"
+	"math"
+	"os"
+	"time"
+
+	"github.com/buildbuddy-io/buildbuddy/server/util/flagutil"
+	"github.com/buildbuddy-io/buildbuddy/server/util/flagutil/yaml"
+	"github.com/buildbuddy-io/buildbuddy/server/util/log"
+	"github.com/prometheus/client_golang/api"
+	"github.com/prometheus/common/model"
+	"golang.org/x/sync/errgroup"
+
+	v1 "github.com/prometheus/client_golang/api/prometheus/v1"
+)
+
+const (
+	// This exit code is returned if the metrics look unhealthy and the rollout should proceed
+	metricsHealthyExitCode = 0
+)
+
+var (
+	config     = flagutil.New("spec", Config{}, "Config to specify which metrics should be monitored by this script.")
+	configPath = flag.String("config_path", "config.yaml", "Path to config file.")
+)
+
+// This script is intended to compare system health for a canary that has been deployed to the rest of the apps
+// If the metrics are healthy, this script will return exit code 0 indicating that the rollout should proceed
+// If the metrics are not healthy for the specified amount of time, the script will return an exit code indicating
+// that the release should be rolled back
+//
+// Pre-requisites:
+//   - You must create a tunnel to prometheus and port-forward in order to run this script with
+//     `kubectl --context="" --namespace=<NAMESPACE> port-forward "<POD_NAME>" --address 0.0.0.0 <FROM_PORT>:<TO_PORT>`
+func main() {
+	flag.Parse()
+	err := yaml.PopulateFlagsFromFile(*configPath)
+	if err != nil {
+		log.Fatalf("Could not read metrics config from %s: %s", *configPath, err)
+	}
+	err = log.Configure()
+	if err != nil {
+		log.Fatalf("Could not configure logger: %s", err)
+	}
+	prometheusClient, err := api.NewClient(api.Config{
+		Address: config.PrometheusAddress,
+	})
+	if err != nil {
+		log.Fatalf("Error creating prometheus client: %v\nDid you run the prometheus port-forward command?", err)
```

Comment:
nit: you can format errors with "%s", doesn't need to be %v

Reply:
Done